packages/web/hooks/useDebouncedCallback.ts

```diff
@@ -0,0 +1,31 @@
+import { useRef, useCallback } from 'react'
+
+/**
+ * Returns a memoized function that will only call the passed function when it hasn't been called for the wait period
+ * @param callback The function to be called
+ * @param wait Wait period after function hasn't been called for
+ * @returns A memoized function that is debounced
+ */
+const useDebouncedCallback = <CallbackArgs extends any[]>(
+  callback: (...args: CallbackArgs) => void,
+  wait: number,
+) => {
+  // Use a ref to store the timeout between renders
+  // and prevent changes to it from causing re-renders
+  const timeout = useRef()
+
+  return useCallback(
+    (...args) => {
+      const later = () => {
+        clearTimeout(timeout.current)
+        callback(...args)
+      }
+
+      clearTimeout(timeout.current)
```

Comment:
So the behavior that's implemented here is that any call to the debounced function will wait at least `wait` ms before invoking the wrapped function, and no invocation will happen if another call to the debounced function happens before invocation of the wrapped func.

This is apparently lodash's definition of "debounce", but I think the behavior we actually want is what lodash calls "throttle", meaning the function will be invoked at most once in a period of `wait` ms, ensuring the last call to the throttled function invoke the wrapped function as soon as `wait` ms has elapsed since the last invocation.

So the logic for that would be something like:
- check the time since the last invocation of the wrapped function
- If it's more than `wait` ms ago, invoke again and update the last-invoked time
- If it's less than `wait` ms ago:
  - Cancel any pending calls (`clearTimeout()`)
  - Create a new pending call scheduled for `wait - (now - lastInvocation)` ms in the future
    - When that pending call does eventually execute, update the last-invoked time
 
This is based on the assumption that in the scrolling case, we do want to continue to call this function while scroll events are happening and continue updating the toolbar position, not wait until all the scrolling action has stopped, we just want to limit the frequency at which that happens